### PR TITLE
Refactor of DEC-939 for readability

### PR DESCRIPTION
**Pull request by danthewolfe**
* Moved some special function code to individual modules.
* Created smaller react components for the lists and document links.

### CODE REVIEW

File: src/components/Document/CatholicDocumentList.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  list() {
+    return SortedParents(this.props.items).map(function(item) {
+      if (item.collection_id == "vatican") {
+        return (<DocumentLink item={item} key={item.id} />);
+      }
+    }.bind(this));
+  }
+
+  render() {
```
Comment (jonhartzler) [inline src/components/Document/CatholicDocumentList.jsx]: We need a title group for the Catholic Documents like human rights.  Even though there is one source Libreria Editrice Vaticana

Comment (danthewolfe) [inline src/components/Document/CatholicDocumentList.jsx]: Any chance of the source being added to the data?

Comment (jonhartzler) [inline src/components/Document/CatholicDocumentList.jsx]: yes i would say there is a chance other than vatican sources will appear we should probably have it do a dynamic list.

**Comment by danthewolfe** [comment]
Changed source to rights_holder:
screen shot 2016-04-29 at 10 18 45 am

**Comment by jonhartzler** [comment]
this is closer.  I think she is planning on adding a field for what she wants specifically to display as the title.  But that is future work.

**Comment by jonhartzler** [comment]
👍

**Pull request merged**